### ISSUE TITLE
Newly created sub emotes are not present in emote picker

### ISSUE THREAD

**Issue by zneix** [Chatterino/chatterino2#2935]
Due to recent changes in Twitch's emote backend (they now use `emotesv2_` IDs) Kraken endpoint fetching all user's emotes does not recognize new user emotes anymore. Thus, we are missing some data and we completely "omit" the new emotes.
Worth noting that since we do a workaround with the emotesets separetly from `USERSTATE` message, this only affects newly added T1 (and possibly partner's T2/T3? not sure...) subscriber emotes.

To fix this, we'd need to get a Helix endpoint for getting authenticated user's emotes which should recognize emotes with `emotesv2_` IDs.
There's already a [uservoice ticket](https://twitch.uservoice.com/forums/310213-developers/suggestions/43599900-get-the-authenticated-user-s-emotes-equivalent-to) for that, but from my experience it'll take a while to get this endpoint in Helix...

**Comment by Mm2PL** [comment]
fucking twitch at it again

**Comment by Magnificus36** [comment]
Is this also why the current new Hype Train emotes, the animal ones [1], don't show up in the picker?
Or is this unrelated and should I open a separate issue?

1: https://help.twitch.tv/s/article/hype-train-guide#emotesntitles

**Comment by zneix** [comment]
> Is this also why the current new Hype Train emotes, the animal ones [1], don't show up in the picker?
> Or is this unrelated and should I open a separate issue?
> 
> 1: [help.twitch.tv/s/article/hype-train-guide#emotesntitles](https://help.twitch.tv/s/article/hype-train-guide#emotesntitles)

tl;dr yea, go for it

Yes, all new Hype Train emotes are using `emotesv2_` format.
Normally, you can fetch emotesets with the new UUID format, but this one set in particular is a special case, because the information about emoteset isn't exposed to us as the emotes in this set are tied to user's account (because of how you unlock them).

We can still get the information about emotes from emotesets that contain emoteset tied to user's emotes, however the new Hype Train emotes use new emoteset format (UUID: `c9f00977-83ce-4391-bef1-525ae806b16f`) and old Kraken endpoint doesn't recognize the new emoteset variant, so there's no way for us to fetch current Hype Train emoteset and emotes in it.

I however managed to fix this issue for most other emotes in #2992 and since the Pull Request closed this issue I guess you can open a new one for Hype Train emotes in particular. and I'll just comment on it with some relevant information.
